### ISSUE TITLE
[Lookup] LookupExtractorFactoryMapContainer will cause jdbc password leak issue

### ISSUE THREAD

**Issue by FaxianZhao** [apache/incubator-druid#8658]
### Description

Coordinator use LookupExtractorFactoryMapContainer instead of LookupExtractorFactoryContainer to store lookup spec, so there is no PasswordProvider to protect jdbc password. 
Anyone could use GET /druid/coordinator/v1/lookups/config/all find them.

**Comment by jon-wei** [comment]
@FaxianZhao Are you using a DefaultPasswordProvider? If so, even if the Coordinator used `LookupExtractorFactoryContainer` instead of `LookupExtractorFactoryMapContainer`, the `DefaultPasswordProvider` would still get serialized and the password would be visible.

You could consider using an EnvironmentPasswordProvider instead, so the password doesn't appear in the spec

**Comment by FaxianZhao** [comment]
> @FaxianZhao Are you using a `DefaultPasswordProvider`? If so, even if the Coordinator used `LookupExtractorFactoryContainer` instead of `LookupExtractorFactoryMapContainer`, the `DefaultPasswordProvider` would still have a visible password in its serialized form.
> 
> You could consider using an `EnvironmentPasswordProvider` in your JDBC extraction namespace instead, so the password doesn't appear

Thanks for your help. If we use `LookupExtractorFactoryContainer` in coordinator, we should load all LookupExtractorFactory extensions when it start. Otherwise, coordinator cannot recognize the right implement.

**Comment by himanshug** [comment]
I agree with @jon-wei  , to protect the password using `EnvironmentVariablePasswordProvider` or your own implementation to `PasswordProvider` would be the right approach.
